### ISSUE TITLE
[camera] Camera preview does not rebuild when device orientation updates.

### ISSUE THREAD

**Issue by BeMacized** [flutter/flutter#87108]
The camera preview widget does not always rebuild when the device orientation updates, causing it to not show the correct orientation. This works fine in the example, but not in a more basic case like the following:
<details>
  <summary>Reproducible example</summary> 

  ```dart
  import 'package:camera/camera.dart';
  import 'package:flutter/material.dart';
  
  List<CameraDescription> cameras = [];
  
  Future<void> main() async {
    WidgetsFlutterBinding.ensureInitialized();
    cameras = await availableCameras();
    runApp(MyApp());
  }
  
  class MyApp extends StatefulWidget {
    @override
    _MyAppState createState() => _MyAppState();
  }
  
  class _MyAppState extends State<MyApp> {
    late CameraController controller;
  
    @override
    void initState() {
      super.initState();
      controller = CameraController(cameras[0], ResolutionPreset.max);
      controller.initialize().then((_) {
        if (!mounted) {
          return;
        }
        setState(() {});
      });
    }
  
    @override
    void dispose() {
      controller.dispose();
      super.dispose();
    }
  
    @override
    Widget build(BuildContext context) {
      if (!controller.value.isInitialized) {
        return Container();
      }
      return MaterialApp(
        home: Scaffold(
          body: Center(
            child: CameraPreview(controller),
          ),
        ),
      );
    }
  }
  ```
</details>

A fix for this has been submitted here: flutter/plugins#4197

**Comment by TahaTesser** [comment]
Hi @BeMacized 
Is this any way related to https://github.com/flutter/flutter/issues/83603? 
Thank you

**Comment by BeMacized** [comment]
Hi @TahaTesser, I too had a hunch it might fix that, but I could sadly still reproduce that one with this fix.

**Comment by TahaTesser** [comment]
The issue is reproducible, thanks!

<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel stable, 2.2.3, on macOS 12.0 21A5284e darwin-x64, locale
    en-GB)
    • Flutter version 2.2.3 at /Users/tahatesser/Code/flutter_stable
    • Framework revision f4abaa0735 (4 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/tahatesser/Code/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/tahatesser/Code/SDK
    • Java binary at: /Users/tahatesser/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Users/tahatesser/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] VS Code (version 1.58.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.24.0

[✓] Connected device (4 available)
    • SM M025F (mobile)      • R9ZR205XX0A                          •
      android-arm    • Android 11 (API 30)
    • iPhone 12 Pro (mobile) • 77C62AB9-8B71-43CF-BCAB-85B0A5B52F85 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64
      • macOS 12.0 21A5284e darwin-x64
    • Chrome (web)           • chrome                               •
      web-javascript • Google Chrome 92.0.4515.107

• No issues found!
```

</details>

**Comment by BeMacized** [comment]
Closing this as the fix in flutter/plugins#4197 has been merged.
